QC/fixed_DFE.py

```diff
@@ -0,0 +1,243 @@
+    optional.add_argument(
+        "--replicates",
+        type=int,
+        default=1,
+        help="Number of replicates to run, taking sum of resulting spectra.",
+    )
+    optional.add_argument(
+        "--population_size",
+        "-N",
+        type=int,
+        default=10000,
+        help="Diploid population size, defaults to 10,000.",
+    )
+    optional.add_argument(
+        "--selection_coefficient",
+        "-s",
+        type=float,
+        default=0.0,
+        help="Selection coefficient for selected mutations, defaults to 0.",
+    )
+    optional.add_argument(
+        "--dominance_coefficient",
+        type=float,
+        default=0.5,
+        help="Dominance coefficient for selected mutations, defaults to 0.5.",
+    )
+    optional.add_argument(
+        "--fraction_selected",
+        "-f",
+        type=float,
+        default=0.7,
+        help="Fraction of new mutations in selected regions that are selected.",
+    )
+    optional.add_argument(
+        "--num_samples",
+        "-n",
+        type=int,
+        default=100,
+        help="Number of (haploid) samples.",
+    )
+    optional.add_argument(
+        "--element_length",
+        "-L",
+        type=int,
+        default=100000,
+        help="Length of the genomic element.",
+    )
+    optional.add_argument(
+        "--flanking_length",
+        type=int,
+        default=0,
+        help="Length of the flanking regions.",
+    )
+    optional.add_argument(
+        "--burn_in", type=int, default=10, help="Burn in factor to pass to slim.",
+    )
+    optional.add_argument(
+        "--scaling_factor",
+        type=int,
+        default=10,
+        help="Scaling factor to pass to slim.",
+    )
+    optional.add_argument(
+        "--plotting",
+        type=str,
+        default=None,
+        help="Plotting output file name. If none given, do not plot.",
+    )
+    return parser
+
+
+def DFE(selCoeff, fracSelected, domCoeff=0.5):
+    """
+    Following the DFE construction from the README in the analysis2 repo.
+    
+    selCoeff: the (unscaled) selection coefficient
+    fracSelected: the proportion of new mutations in selected element that are
+        selected. 1 - fracSelected will be neutral.
+    """
+    neutral = stdpopsim.ext.MutationType()
+    h = domCoeff
+    selected = stdpopsim.ext.MutationType(
+        dominance_coeff=domCoeff, distribution_type="f", distribution_args=[selCoeff]
+    )
+    return {
+        "mutation_types": [neutral, selected],
+        "proportions": [1 - fracSelected, fracSelected],
+    }
+
+
+def set_up_contig(args, species):
+    """
+    Sets up the contig with the selected element and flanking region.
+    """
+    F = args.flanking_length
+    L = args.element_length
+    total_length = 2 * F + L
+    contig = species.get_contig(length=total_length)
+    # need to clear genome mutation types before adding them ???
+    contig.clear_genomic_mutation_types()
```

Comment:
This is very confusing to me, I think we need to do something about it!

Reply:
This is related to popsim-consortium/stdpopsim#973, right? I agree that this would be a good think to clean up in the API.